contracts/liquidator/Liquidator.sol

```diff
@@ -28,114 +35,212 @@ contract Liquidator is IUniswapV3FlashCallback, PeripheryImmutableState, Periphe
-        bool reversedPair;
     }
 
-    using LowGasSafeMath for uint256;
-    using LowGasSafeMath for int256;
+    struct UniswapPoolConfig {
+        address asset;
+        bool isLowLiquidity;
+        uint24 fee;
+    }
+
+    /** Liquidator configuration constants **/
+
+    /// @notice The scale for asset price calculations
+    uint256 public constant QUOTE_PRICE_SCALE = 1e6;
 
+    /// @notice Uniswap standard pool fee, used if custom fee is not specified for the pool
+    uint24 public constant DEFAULT_POOL_FEE = 500;
+
+    /// @notice Uniswap router used for token exchange
     ISwapRouter public immutable swapRouter;
+
+    /// @notice Compound Comet protocol
     CometInterface public immutable comet;
 
-    uint24 public constant defaultPoolFee = 500;
-    mapping(address => uint24) public poolFees;
+    /// @notice Minimum available amount for liquidation in USDC (base token)
+    uint256 public immutable liquidationThreshold;
 
+    /// @notice The address of WETH asset
+    address public immutable weth;
+
+    UniswapPoolConfig[] public poolConfigs;
+
+    /** Uniswap pools properties **/
+    mapping(address => uint24) public poolFees;
+    mapping(address => bool) public isLowLiquidity;
+
+    /**
+     * @notice Construct a new liquidator instance
+     * @param _swapRouter The Uniswap V3 Swap router address
+     * @param _comet The Compound V3 Comet instance address
+     * @param _factory The Uniswap V3 pools factory instance address
+     * @param _WETH9 The WETH address
+     * @param _poolConfigs The configurations of Uniswap asset pools
+     **/
     constructor(
         ISwapRouter _swapRouter,
         CometInterface _comet,
         address _factory,
         address _WETH9,
-        address[] memory _assets,
-        uint24[] memory _poolFees
+        uint256 _liquidationThreshold,
+        UniswapPoolConfig[] memory _poolConfigs
     ) PeripheryImmutableState(_factory, _WETH9) {
-        require(_assets.length == _poolFees.length, "Wrong input");
-
         swapRouter = _swapRouter;
         comet = _comet;
+        weth = _WETH9;
+        liquidationThreshold = _liquidationThreshold;
 
-        // Set the desirable pool fees for assets
-        for (uint i = 0; i < _assets.length; i++) {
-            address asset = _assets[i];
-            uint24 poolFee = _poolFees[i];
-            poolFees[asset] = poolFee;
+        poolConfigs = _poolConfigs;
+    }
+
+    /**
+     * @dev Returns Uniswap pool config for given asset
+     */
+    function getPoolConfigForAsset(address asset) internal view returns(UniswapPoolConfig memory) {
+        for (uint8 i = 0; i < poolConfigs.length; i++) {
+            UniswapPoolConfig memory poolConfig = poolConfigs[i];
+            if (poolConfig.asset == asset) {
+                return poolConfig;
+            }
         }
+
+        // If asset is not found, proceed with default pool config
+        return UniswapPoolConfig({
+            fee: 500,
```

Comment:
I think this is where you would want to use `DEFAULT_POOL_FEE`.

Reply:
great catch, I will update that.